File: library-backend/src/main/java/com/adrian/library/copy/CopyService.java
package com.adrian.library.copy;

import com.adrian.library.borrowing.BookAlreadyBorrowedException;
import com.adrian.library.reservation.BookAlreadyReservedException;
import com.adrian.library.user.UserLacksActionPointsException;

import java.util.List;

public interface CopyService {
    List<Copy> getAll();
    List<Copy> getByEditionId(Integer editionId);
    Copy get(Integer id);
    Copy create(Copy copy) throws UserLacksActionPointsException, BookAlreadyReservedException, BookAlreadyBorrowedException;
    Copy update(Copy copy);
    void delete(Integer id);
}

File: library-frontend/src/services/CopyService.js
import axios from 'axios'

const COPY_URL = 'http://localhost:8080/copies';

class CopyService {

  getAll() {
    const response = axios({
      method: 'get',
      url: COPY_URL,
      headers: { 
        'Authorization': `Bearer ${localStorage.getItem('user-token')}`
      },
    })
    .catch(function (error) {
      console.log(error);
    });

    return response;
  }

  getByEditionId(editionId) {
    const response = axios({
      method: 'get',
      url: COPY_URL + "/edition/" + editionId,
      headers: { 
        'Authorization': `Bearer ${localStorage.getItem('user-token')}`
      }
    })
    .catch(function (error) {
      console.log(error);
    });
    
    return response;
  }

  create(copy) {
    console.log(copy);
    const response = axios({
      method: 'post',
      url: COPY_URL,
      data: copy,
      headers: { 
        'Authorization': `Bearer ${localStorage.getItem('user-token')}`
      },
    })
    .catch(function (error) {
      console.log(error);
    });

    return response;
  }

  update(copy) {
    const response = axios({
      method: 'put',
      url: COPY_URL,
      data: copy,
      headers: { 
        'Authorization': `Bearer ${localStorage.getItem('user-token')}`
      },
    })
    .catch(function (error) {
      console.log(error);
    });

    return response;
  }
}

export default new CopyService();